### PR TITLE
preserve ctx when dialing unix socket for desktop

### PR DESCRIPTION
**Pull request by James-Pickett**
we see a lot of errors communicating over desktop socket, I don't know if properly passing the context will help with that, but seems correct either way

### CODE REVIEW

**Review by zackattack01 — DISMISSED** [review]

File: ee/desktop/user/client/dialer_posix.go [hunk 1]
```diff
@@ -9,7 +9,8 @@ import (
 )
 
 func dialContext(socketPath string) func(_ context.Context, _, _ string) (net.Conn, error) {
-	return func(_ context.Context, _, _ string) (net.Conn, error) {
-		return net.Dial("unix", socketPath)
+	return func(ctx context.Context, _, _ string) (net.Conn, error) {
+		d := net.Dialer{}
+		return d.DialContext(ctx, "unix", socketPath)
```
Comment (James-Pickett) [inline ee/desktop/user/client/dialer_posix.go:14]: wonder if we should reuse dialer?

```suggestion
        d := net.Dialer{}
	return func(ctx context.Context, _, _ string) (net.Conn, error) {
		return d.DialContext(ctx, "unix", socketPath)
```

Comment (James-Pickett) [reply ee/desktop/user/client/dialer_posix.go]: updated to do the same thing the net pkg does

File: ee/desktop/user/client/dialer_posix.go [hunk 2]
```diff
@@ -9,7 +9,8 @@ import (
 )
 
 func dialContext(socketPath string) func(_ context.Context, _, _ string) (net.Conn, error) {
```
Comment (directionless) [inline ee/desktop/user/client/dialer_posix.go:11]: Does the return need `ctx`? I'm not really sure how named vars work here

**Review by directionless — COMMENTED** [review]
Seems reasonable. one question

File: ee/desktop/user/client/dialer_posix.go (hunk 2 above)
Comment (James-Pickett) [reply ee/desktop/user/client/dialer_posix.go]: I'm not sure, so I added just incase, it shouldn't hurt anything

Comment (James-Pickett) [reply ee/desktop/user/client/dialer_posix.go:11]: actually, nvm, it's not needed

https://go.dev/play/p/BPUDbYv-slf

**Review by directionless — APPROVED** [review]

File: ee/desktop/user/client/dialer_posix.go (hunk 2 above)
Comment (directionless) [reply ee/desktop/user/client/dialer_posix.go:11]: Thank you for checking

**Review by RebeccaMahany — APPROVED** [review]

**Pull request merged**